conans/test/functional/graph_lock/graph_lock_ci_test.py

```diff
@@ -58,232 +63,121 @@ def package_info(self):
+                client_aux.run("install %s --build=%s --lockfile=conan.lock" % (ref, ref))
                 lock_fileaux = client_aux.load(LOCKFILE)
                 client.save({"new_lock/%s" % LOCKFILE: lock_fileaux})
-                client.run("graph update-lock . new_lock")
+                client.run("lock update conan.lock new_lock/conan.lock")
 
-            client.run("graph build-order . --build=cascade")
+            client.run("lock build-order conan.lock --json=bo.json")
             lock_fileaux = client.load(LOCKFILE)
-            output = str(client.out).splitlines()[-1]
-            to_build = eval(output)
+            to_build = json.loads(client.load("bo.json"))
 
         new_lockfile = client.load(LOCKFILE)
-        client.run("install PkgD/0.1@user/channel --lockfile")
+        client.run("install PkgD/0.1@user/channel --lockfile=conan.lock")
         self.assertIn("PkgC/0.1@user/channel: DEP FILE PkgB: ByeB World!!", client.out)
         self.assertIn("PkgD/0.1@user/channel: DEP FILE PkgB: ByeB World!!", client.out)
-        client.run("upload * --all --confirm")
 
-        client.save({LOCKFILE: initial_lock_file})
-        client.run("remove * -f")
-        client.run("install PkgD/0.1@user/channel --lockfile")
+        client.save({LOCKFILE: initial_lockfile})
+        client.run("install PkgD/0.1@user/channel --lockfile=conan.lock")
         self.assertIn("PkgC/0.1@user/channel: DEP FILE PkgB: HelloB", client.out)
         self.assertIn("PkgD/0.1@user/channel: DEP FILE PkgB: HelloB", client.out)
 
         client.save({LOCKFILE: new_lockfile})
-        client.run("remove * -f")
-        client.run("install PkgD/0.1@user/channel --lockfile")
+        client.run("install PkgD/0.1@user/channel --lockfile=conan.lock")
         self.assertIn("PkgC/0.1@user/channel: DEP FILE PkgB: ByeB World!!", client.out)
         self.assertIn("PkgD/0.1@user/channel: DEP FILE PkgB: ByeB World!!", client.out)
 
-    def test_version_ranges(self):
-        conanfile = textwrap.dedent("""
-            from conans import ConanFile, load
-            import os
-            class Pkg(ConanFile):
-                {requires}
-                exports_sources = "myfile.txt"
-                keep_imports = True
-                def imports(self):
-                    self.copy("myfile.txt", folder=True)
-                def package(self):
-                    self.copy("*myfile.txt")
-                def package_info(self):
-                    self.output.info("SELF FILE: %s"
-                        % load(os.path.join(self.package_folder, "myfile.txt")))
-                    for d in os.listdir(self.package_folder):
-                        p = os.path.join(self.package_folder, d, "myfile.txt")
-                        if os.path.isfile(p):
-                            self.output.info("DEP FILE %s: %s" % (d, load(p)))
-                """)
+    def test_version_ranges_partial(self):
```

Comment:
Please @czoido have a look to the flow proposed by this test and let me know your feedback

Reply:
@memsharded,
I really like the new flow, I think it's more intuitive now and that is going yo simplify things in CI flows.